src/Features/CSharp/Portable/InlineHints/CSharpInlineParameterNameHintsService.cs

```diff
@@ -55,13 +57,13 @@ public CSharpInlineParameterNameHintsService()
                     continue;
 
                 var parameter = argument.DetermineParameter(semanticModel, cancellationToken: cancellationToken);
-                buffer.Add((argument.Span.Start, parameter, GetKind(argument.Expression)));
+                buffer.Add((argument.Span.Start, argument, parameter, GetKind(argument.Expression)));
```

Comment:
you only add the argument to get the expression/identifier.  instead of that, just add the identifier-text here (if that's what you have).  and teh caller can do all teh filtering out trivially.

Reply:
that's true, changed